product-tours/tour-tf-keras/Tour-with-TensorFlow-and-Keras.ipynb

```diff
@@ -0,0 +1,2761 @@
+    "## Step 7: Stop experiment at the end"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment"
+    ]
+   },
+   "source": [
+    "This statement is needed only in Notebooks. In scripts experiment is closed automatically at the end."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "neptune.stop()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "tests"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# tests\n",
+    "## check logs\n",
+    "correct_logs = ['batch_loss', 'batch_accuracy', 'epoch_loss', 'epoch_accuracy', 'test_loss', 'test_accuracy',\n",
+    "                'epoch_val_loss', 'epoch_val_accuracy']\n",
+    "\n",
+    "if set(exp.get_logs().keys()) != set(correct_logs):\n",
+    "    raise ValueError('incorrect metrics')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "UphOuRMm5sZs",
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "# More logging options"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment"
+    ]
+   },
+   "source": [
+    "Below we present more logging options. Flow of the example is similar to the basic one. The difference is that here we present more logging options.\n",
+    "\n",
+    "* logging experiments parameters,\n",
+    "* logging data version,\n",
+    "* model architecture summary,\n",
+    "* weights of the trained model,\n",
+    "* predictions on the test set as table,\n",
+    "* visualizations of the model performance,\n",
+    "* training data sample."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "## Install additional dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "DO7yawFVxVIt",
+    "tags": [
+     "installation"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "! pip install --quiet scikit-plot==0.3.7 matplotlib==3.3.3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "DO7yawFVxVIt",
+    "tags": [
+     "installation",
+     "library_updates"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "! pip install --quiet scikit-plot matplotlib --upgrade"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "## Import Libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "import hashlib\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
```

Comment:
I think we should import those libraries where we need them -> somebody may want to jump and see what is needed to do X and then having it in one place is great

Reply:
ok, changed.